odxtools/database.py

```diff
@@ -89,31 +85,35 @@ def _process_xml_tree(self, root: ElementTree.Element) -> None:
 
         self.model_version = model_version
 
-        dlc = root.find("DIAG-LAYER-CONTAINER")
-        if dlc is not None:
-            dlcs.append(DiagLayerContainer.from_et(dlc, OdxDocContext(model_version, [])))
-
-        # In ODX 2.0 there was only COMPARAM-SPEC. In ODX 2.2 the
-        # content of COMPARAM-SPEC was moved to COMPARAM-SUBSET
-        # and COMPARAM-SPEC became a container for PROT-STACKS and
-        # a PROT-STACK references a list of COMPARAM-SUBSET
-        cp_subset = root.find("COMPARAM-SUBSET")
-        if cp_subset is not None:
-            comparam_subsets.append(
-                ComparamSubset.from_et(cp_subset, OdxDocContext(model_version, [])))
-
-        cp_spec = root.find("COMPARAM-SPEC")
-        if cp_spec is not None:
-            if model_version < Version("2.2"):
-                comparam_subsets.append(
-                    ComparamSubset.from_et(cp_spec, OdxDocContext(model_version, [])))
-            else:  # odx >= 2.2
-                comparam_specs.append(
-                    ComparamSpec.from_et(cp_spec, OdxDocContext(model_version, [])))
-
-        self._diag_layer_containers.extend(dlcs)
-        self._comparam_subsets.extend(comparam_subsets)
-        self._comparam_specs.extend(comparam_specs)
+        child_elements = list(root)
+        if len(child_elements) != 1:
+            odxraise("Each ODX document must contain exactly one category.")
+
+        category_et = child_elements[0]
+        category_sn = odxrequire(category_et.findtext("SHORT-NAME"))
+
+        match category_et.tag:
```

Comment:
the `match` statement is for "structural pattern matching" ([pep](https://peps.python.org/pep-0622/), I guess I'm too stupid to understand what that is), and AFAIK it is not more performant than a simple `if ... elif` chain. Please replace...